src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -172,77 +212,162 @@ private BoundPattern BindPropertyPattern(PropertyPatternSyntax node, BoundExpres
             return new BoundPropertyPattern(node, type, boundPatterns, hasErrors: hasErrors);
         }
 
-        private ImmutableArray<BoundSubPropertyPattern> BindSubPropertyPatterns(PropertyPatternSyntax node, TypeSymbol type, DiagnosticBag diagnostics)
+        private ImmutableArray<BoundSubPropertyPattern> BindSubPropertyPatterns(
+            PropertyPatternSyntax node,
+            TypeSymbol type,
+            DiagnosticBag diagnostics)
         {
             var result = ArrayBuilder<BoundSubPropertyPattern>.GetInstance();
-            foreach (var syntax in node.PatternList.SubPatterns)
+            foreach (var e in node.SubPatterns)
             {
-                var propName = syntax.Left;
-                BoundPattern pattern;
-                HashSet<DiagnosticInfo> useSiteDiagnostics = null;
-                LookupResultKind resultKind;
-                Symbol property = FindPropertyOrFieldByName(type, propName, out resultKind, ref useSiteDiagnostics);
-                if ((object)property != null)
-                {
-                    bool hasErrors = false;
-                    if (property.IsStatic)
-                    {
-                        Error(diagnostics, ErrorCode.ERR_ObjectProhibited, propName, property);
-                        hasErrors = true;
-                    }
-                    else
-                    {
-                        diagnostics.Add(node, useSiteDiagnostics);
-                    }
+                var syntax = e as IsPatternExpressionSyntax;
+                var identifier = syntax?.Expression as IdentifierNameSyntax;
+                if (identifier == null) throw ExceptionUtilities.UnexpectedValue(syntax?.Expression.Kind() ?? e.Kind());
```

Comment:
The spec is in flux, and this is an open issue. I expect we might support  `[0]` in place of an identifier for indexing the object, and an error expression syntax node when there is a syntax error inside.